azurelinuxagent/ga/exthandlers.py

```diff
@@ -684,7 +684,6 @@ def __init__(self, ext_handler, protocol):
         self.operation = None
         self.pkg = None
         self.pkg_file = None
-        self.is_upgrade = False
```

Comment:
I'm very curious to know as to why this property was added in the first place?

Reply:
[This PR](https://github.com/Azure/WALinuxAgent/pull/430/files#diff-f8c44add0bc67ec987eff75282ec2308R203) brought it in. It seems the flag was only used to intentionally retry the same goal state if it's an upgrade scenario. I believe the intention was to add a retry in case of transient failures during upgrade, but we've since seen how messy and costly it is when it's a non-transient upgrade failure (hint: OMS), so I don't it's a good trade-off to keep this logic.

Reply:
I cant think of a scenario where this might hit, but I'm afraid of customers who might've taken dependency on this "behavior" where we keep trying to retry goalstate if there's an extension upgrade available.
Anyways I feel even if someone did take a dependency then they can modify the behavior to the correct behavior

Reply:
@larohra: By _taking dependency on this "behavior"_, do you mean writing extensions assuming that GA would be retrying till the end of time?

@pgombar, we have to make it clear in our documentation as well to make it clear and concrete and we would only try once, if not already in our documentation.

Reply:
yeah basically. Like what I meant was since we even had a test case that was pretty much trying to check if the disable failed scenario would recover itself without a new incarnation change, maybe the previous owners told the extension publishers to not worry about transient issues as they would be "auto-resolved" eventually due to which they dont have enough retry logic in their code. 

@vrdmr I agree with your point about double checking the documentation and making it very explicit that we would only try it once per goalstate.